### DISCUSSION TITLE
Image caching: configure cache sizes

### DISCUSSION THREAD

**Discussion by fiznool** [react-native-community/discussions-and-proposals#663] [Ideas]
The built-in `<Image />` component is backed by a solid caching layer, meaning that frequently-used images will be cached for use later. The cache respects standard HTTP caching headers, and therefore works very well without any up-front configuration from the app side.

That being said, out of the box React Native applies [cache limits](https://github.com/facebook/react-native/blob/a4978823841177ff70959e28ed8343b78fb14c20/packages/react-native/Libraries/Image/RCTImageCache.m#L21C1-L22) that may not be the most appropriate for individual apps. Image-heavy apps, for example, may wish to increase these cache limits to more aggresively cache images for (e.g.) better offline support.

It would be nice to include a mechanism for changing these limits on a per-app basis.

**Comment by sahariar-safin** [comment]
I agree that it would be nice to include a mechanism for changing the cache limits on a per-app basis. This would allow developers to fine-tune the caching behavior of their apps to meet their specific needs.

**Here are a few ideas for how this could be implemented:**

- The cache limits could be exposed as configuration options in the React Native CLI. This would make it easy for developers to set the limits when they create their apps.
- The cache limits could be exposed as properties on the Image component. This would allow developers to change the limits on a per-component basis.
- The cache limits could be stored in a configuration file. This would allow developers to override the default limits without having to modify the code of their app.

I think the best approach would depend on the specific needs of the React Native community. However, I believe that it is important to provide developers with the ability to control the caching behavior of their apps.

**Here are some additional details about how the cache limits could be implemented:**

- The cache size limit could be specified in bytes. This would allow developers to set a precise limit on the amount of disk space that can be used for caching images.
- The total cost limit could be specified in bytes or in terms of the number of images that can be cached. This would allow developers to trade off between the amount of disk space that is used for caching images and the number of images that can be cached.
- The cache limits could be applied to all images or to specific images. This would allow developers to fine-tune the caching behavior of their apps.
-

I believe that these features would make React Native a more powerful and flexible platform for developing image-heavy apps.
